src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -45,7 +47,7 @@ public ErrorLogger(Stream stream, string toolName, string toolFileVersion, Versi
 
         private void WriteToolInfo(string name, string fileVersion, Version assemblyVersion)
         {
-            _writer.WriteObjectStart("toolInfo");
+            _writer.WriteObjectStart("tool");
             _writer.Write("name", name);
             _writer.Write("version", assemblyVersion.ToString(fieldCount: 3));
```

Comment:
The convention I've followed in the SARIF SDK is to emit the four-part assembly version to the 'version' field. Then emit the three-part version as the 'semanticVersion' field. Then, we emit the assembly file version but only if it differs from the assembly version.